go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -124,10 +124,13 @@ func isUnrecoverableError(err error) bool {
 	if err == nil {
 		return false
 	}
-	sqlErr, isSQLErr := err.(*mysql.SQLError)
+	sqlErr, isSQLErr := mysql.NewSQLErrorFromError(err).(*mysql.SQLError)
 	if !isSQLErr {
 		return false
 	}
+	if sqlErr.Num == mysql.ERUnknownError {
+		return false
+	}
```

Comment:
I think this is only needed here because isSQLErr will now always be true, no?

Reply:
Answered in main comment